client/scala/src/main/scala/ai/verta/blobs/dataset/S3Location.scala

```diff
@@ -0,0 +1,39 @@
+package ai.verta.blobs.dataset
+
+import java.net.{URL, URI}
+
+import scala.util.{Try, Success, Failure}
+
+/** A location in S3
+ *  To initialize, pass in an S3 path and (optionally) a version ID:
+ *  {{{
+ *  val S3Location: Try[S3Location] = S3Location("some-path-1")
+ *  }}}
+ */
+class S3Location(
```

Comment:
```suggestion
case class S3Location(
```
?

Reply:
Making S3Location a case class is gonna create conflict when calling:
```
S3Location("some-bucket-name")
```
because it's not gonna know whether to pick which of the following:
```
apply(bucketName, key = None, versionId = None): S3Location // auto-provided
apply(path, versionID = None): Try[S3Location] // mine
```
We need my apply method to check the validity of Path.